datasets.py

```diff
@@ -919,3 +920,110 @@ def from_path(self, file_path):
         anns = [image_anns, gender_info, [country], file_path, scene_group]
 
         return image, anns
+
+class CityScapes(data.Dataset):
+
+    def __init__(self, transform): 
+        self.transform = transform
+        self.img_folder = '/Users/home/Desktop/research/data/cityscapes/gtFine_trainvaltest/gtFine/train'
+
+        # directory storing gps information
+        self.gps_folder = '/Users/home/Desktop/research/data/cityscapes/vehicle_trainvaltest/vehicle/train'
+
+        # store all of the city names in array [aachen, bochum, etc]
+        self.city_names = os.listdir(self.gps_folder)
+        self.city_names.remove('.DS_Store')
+
+        # Adds the title of the image as its ID 
+        # (e.g. aachen/aachen_000000_000019_gtFine_color.png = aachen/aachen_000000_000019)
+        self.image_ids = []
+        for city in self.city_names:
+            filename_path = os.path.join(self.gps_folder, city)
+            city_filenames = os.listdir(filename_path)
+            self.image_ids = self.image_ids +  [os.path.join(city, name.split("_vehicle")[0]) for name in city_filenames]
+        print("done with ids (1/2)")
+
+        with open("/Users/home/Downloads/stanford-nh891yz3147-geojson.json") as f:
+            self.geo_boundaries = json.load(f)
+
+        # TODO: categories work
```

Comment:
what does this mean?

Reply:
this is loading in the geography boundaries relevant to the dataset. here for example, the geography boundary is the administrative regions of germany